src/order/antisymmetrization.lean

```diff
@@ -0,0 +1,175 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import order.hom.basic
+
+/-!
+# Turning a preorder into a partial order
+
+This file allows to make a preorder into a partial order by quotienting out the elements `a`, `b`
+such that `a ≤ b` and `b ≤ a`.
+
+`antisymmetrization` is a functor from `Preorder` to `PartialOrder`. See `Preorder_to_PartialOrder`.
+
+## Main declarations
+
+* `le_equiv`: The antisymmetrization relation. `le_equiv a b` means that `a` and `b` are less than
+  each other.
+* `antisymmetrization α`: The quotient of `α` by `le_equiv`. Even when `α` is just a preorder,
+  `antisymmetrization α` is a partial order.
+-/
+
+open order_dual
+
+variables {α β : Type*}
+
+section has_le
+variables [has_le α]
+
+/-- The antisymmetrization relation. -/
+def le_equiv (a b : α) : Prop := a ≤ b ∧ b ≤ a
+
+lemma le_equiv.symm {a b : α} : le_equiv a b → le_equiv b a := and.symm
+
+lemma le_equiv_refl [is_refl α (≤)] (a : α) : le_equiv a a := ⟨refl _, refl _⟩
+
+lemma le_equiv.trans [is_trans α (≤)] {a b c : α} (hab : le_equiv a b) (hbc : le_equiv b c) :
+  le_equiv a c :=
+⟨trans hab.1 hbc.1, trans hbc.2 hab.2⟩
+
+instance le_equiv.decidable_rel [@decidable_rel α (≤)] : @decidable_rel α le_equiv :=
+λ _ _, and.decidable
+
+end has_le
+
+section preorder
+variables [preorder α] [preorder β] {a b : α}
+
+lemma le_equiv.image {f : α → β} (hf : monotone f) {a b : α} (h : le_equiv a b) :
+  le_equiv (f a) (f b) :=
+⟨hf h.1, hf h.2⟩
+
+variables (α)
+
+/-- The antisymmetrization relation as an equivalence relation. -/
+@[simps] def le_equiv.setoid : setoid α :=
+⟨le_equiv, le_equiv_refl, λ _ _, le_equiv.symm, λ _ _ _, le_equiv.trans⟩
+
+/-- The partial order derived from a preorder by making pairwise comparable elements equal. This is
+the quotient by `λ a b, a ≤ b ∧ b ≤ a`. -/
+def antisymmetrization : Type* := quotient $ le_equiv.setoid α
```

Comment:
I would be inclined to put this in the `order` namespace, especially since there was a recent PR about symmetrized algebras.

Reply:
What about now,  given that it's generalized to an arbitrary relation?